plugins/notification-mappers/src/index.ts

```diff
@@ -0,0 +1,190 @@
+    objectClass: spaceInfo._class,
+    _id: generateId(),
+    _class: core.class.TxUpdateDoc,
+    space: core.space.Tx,
+    modifiedBy: tx.modifiedBy,
+    modifiedOn: tx.modifiedOn,
+    createOn: tx.createOn,
+    operations: {
+      lastModified: tx.modifiedOn
+    }
+  }
+
+  await options.storage.tx(updateTx)
+}
+
+function createSpaceNotification<T extends Space> (
+  ctx: TxCreateDoc<T> | TxUpdateDoc<T>,
+  options: MappingOptions,
+  member: Ref<Account>
+): SpaceNotifications {
+  return {
+    _id: `spaceNotification-${ctx.objectId}-${member}` as Ref<SpaceNotifications>,
+    lastRead: ctx.modifiedOn,
+    createOn: ctx.createOn,
+    descriptorId: options.descriptor._id,
+    space: member.toString() as Ref<Space>,
+    objectClass: ctx.objectClass,
+    objectId: ctx.objectId,
+    modifiedOn: ctx.modifiedOn,
+    _class: notification.class.SpaceNotifications,
+    modifiedBy: ctx.modifiedBy,
+    notificatedObjects: [],
+    objectLastReads: new Map<Ref<Doc>, Timestamp>()
+  }
+}
+
+function createSpaceInfo<T extends Space> (
+  ctx: TxCreateDoc<T> | TxUpdateDoc<T>,
+  descriptorId: Ref<DerivedDataDescriptor<Doc, DerivedData>>
+): SpaceInfo {
+  const data: SpaceInfo = {
+    _class: notification.class.SpaceInfo,
+    objectId: ctx.objectId,
+    objectClass: ctx.objectClass,
+    _id: `dd-spaceInfo-${ctx.objectId}` as Ref<SpaceInfo>,
+    modifiedBy: ctx.modifiedBy,
+    modifiedOn: ctx.modifiedOn,
+    createOn: ctx.createOn,
+    space: ctx.objectId,
+    descriptorId: descriptorId,
+    lastModified: ctx.modifiedOn
+  }
+  return data
+}
+
+export default (): void => {
+  registerMapper(notification.mappers.SpaceInfo, {
+    map: async (tx: Tx, options: MappingOptions): Promise<SpaceInfo[]> => {
+      if (tx._class === core.class.TxCreateDoc) {
+        const ctx = tx as TxCreateDoc<Doc>
+        if (options.hierarchy.isDerived(ctx.objectClass, core.class.Space)) {
+          return [createSpaceInfo(ctx as TxCreateDoc<Space>, options.descriptor._id)]
+        } else if (
+          ctx.objectSpace !== core.space.Model &&
+          !options.hierarchy.isDerived(ctx.objectClass, core.class.DerivedData)
+        ) {
+          await updateSpaceInfo(ctx, options)
+        }
+      }
+
+      if (tx._class === core.class.TxUpdateDoc) {
+        const ctx = tx as TxUpdateDoc<Doc>
+        if (options.hierarchy.isDerived(ctx.objectClass, core.class.Space)) {
+          const result = await options.storage.findAll(notification.class.SpaceInfo, { objectId: ctx.objectId })
+          if (result.length === 0) {
+            return [createSpaceInfo(ctx as TxUpdateDoc<Space>, options.descriptor._id)]
+          }
+          return result
+        } else if (
+          ctx.objectSpace !== core.space.Model &&
+          !options.hierarchy.isDerived(ctx.objectClass, core.class.DerivedData)
+        ) {
+          await updateSpaceInfo(ctx, options)
+        }
+      }
+
+      if (tx._class === core.class.TxRemoveDoc) {
```

Comment:
Do you plan to handle Space Removal case?

Reply:
I not need handle this because mapper call in only update|create tx. I will remove this